doc/elements-confidential-transactions.md

```diff
@@ -0,0 +1,480 @@
+and amounts, while allowing a verifier to prove that the sum of a
+transaction’s inputs is equal to the sum of its outputs plus the
+transaction fee, without needing to see the actual amounts.
+
+### Range Proofs
+
+When constructing a confidential transaction using Pedersen
+commitments, it is possible to introduce an output with a negative
+value, thereby causing coin inflation in other outputs. To prevent
+this, a cryptographic range proof is included in each blinded
+transaction output. A range proof allows a verifier to prove that the
+output value lies within a specified non-negative range, without
+having to see the actual value.
+
+### Surjection Proofs
+
+When assets are blinded in a transaction, a verifier cannot see which
+input assets are sent to which outputs. A Surjection proof [2] allows
+a verifier to prove that an output’s asset appears in at least one
+input, without revealing the actual asset type. (The mapping from
+input assets to the output asset must be an “onto” function, or a
+“surjection”). Every blinded transaction output has a Surjection
+proof.
+
+### Elliptic Curve Diffie-Hellman (ECDH)
+
+The receiver and the sender need a shared secret to blind and unblind
+the outputs and to create the range proof. ECDH key exchange is used
+to create the shared secret. When generating a Confidential Address,
+the receiver creates a private blinding key and embeds the public key
+in the address. The sender generates an ephemeral keypair, and with
+the address’ public blinding key, uses ECDH to generate a shared
+secret. The sender’s key for the ECDH key exchange is written into the
+“ConfidentialNonce” field of the transaction output.
+
+See the “Confidential Assets” paper [3] for more detailed information
+and references.
+
+## Transaction Layout and Size Considerations
+
+The layout and fields of a Confidential Transaction are described in
+[elements-tx-format.md](./elements-tx-format.md).
+
+Confidential Transactions are significantly larger than
+non-confidential transactions due to the addition of the range and
+surjection proofs.  Each blinded output adds about 4k bytes to the
+transaction. As the proofs are stored in output witness fields, each
+blinded output adds about 1k vBytes to the virtual size of the
+transaction (please see https://en.bitcoin.it/wiki/Weight_units for
+information about transaction size measurements).
+
+### Range Proofs
+
+A range proof proves that a value lies within a specific range. The
+exact range is determined by the parameters of the range proof. A
+tradeoff exists between the size of the range, and the size of the
+proof’s serialization in the transaction output.  A larger range
+requires a larger proof. In Elements, the default size of the range
+proof is 52 bits, allowing a value up to 2^52-1 satoshis to be
+represented. In Elements, it is possible to create more than 21
+million satoshis of an asset over multiple issuances, but it is not
```

Comment:
Is this correct? The per-transaction limit is not 21 million right? Since, in bitcoin there are `21,000,000 btc x 100,000,000 sats / btc` total satoshis?

Reply:
thanks, fixed in later commit

Reply:
Looks like it still reads `21 million satosahis`. Was this addressed?